src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.DefaultItems.props

```diff
@@ -37,4 +37,8 @@ Copyright (c) .NET Foundation. All rights reserved.
     <PackageReference Include="Microsoft.NETCore.App" Version="$(RuntimeFrameworkVersion)" IsImplicitlyDefined="true" />
   </ItemGroup>
 
+  <ItemGroup Condition=" '$(DisableImplicitFrameworkReferences)' != 'true' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and !$(RuntimeFrameworkVersion.StartsWith('1.'))">
+    <PackageReference Include="Microsoft.NETCore.DotNetAppHost" Version="$(RuntimeFrameworkVersion)" IsImplicitlyDefined="true" />
```

Comment:
https://github.com/dotnet/core-setup/issues/1722

> but that does not affect the change here in SDK

It does affect the change here in the SDK.  If we configure the packages efficiently, the tooling doesn't need all this logic in it.  See the issue above.  In this change we are doing things like 

* Delete the `dotnet.exe` file from the published files.
* First look for `apphost.exe`, if that isn't found fall back to `dotnet.exe`.

The tooling logic can be a lot simpler.